### PR TITLE
fix(cli): resume checkpoint

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Summary

- Fixed `last.ckpt` freezing at the best epoch: Lightning's `save_last` only mirrors monitor-driven saves, so runs whose `val/loss` peaked early had no usable resume point; a dedicated monitor-less `ModelCheckpoint` now saves it every epoch.
- `--resume-checkpoint` continues the checkpoint's source MLflow run (same run ID, checkpoint dir, and metric curves) via the adjacent `run_metadata.json`; fails at launch on missing metadata or config mismatch.
- Added `--new-run` to explicitly fork a resumed training into a fresh MLflow run.
- Resuming with a modified config now works: params stay append-only (MLflow immutability), changed keys are warned per key and recorded in a `param_drift` run tag.
- New `ConfigAuthoritativeStateMixin` for callbacks: on restore, state keys that are also constructor parameters take the configured value (e.g. early-stopping patience), runtime state like `wait_count` is restored unchanged; no field names hardcoded.
- `train.py` logs the resume checkpoint's epoch/step before fitting, so a stale checkpoint is visible immediately.
- Resume paths are resolved to absolute and validated in the CLI.

TODO: logger abstraction (remove MLFlow hard dependency).

## Checklist

- [ ] You've checked our [contribution overview](https://tier4.github.io/autoware-ml/contributing/contribution-overview/).
- [ ] Your PR follows our [pull request guidelines](https://tier4.github.io/autoware-ml/contributing/pull-request-guidelines/).

### CODE REVIEW

File: autoware_ml/tests/framework/test_resume_support.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        assert callback.patience == 15
+        assert caplog.text == ""
+
+
+class TestResumedParamLogging:
+    def _logger_with_params(self, params: dict[str, str]) -> MagicMock:
+        trainer_logger = MagicMock(spec=MLFlowLogger)
+        trainer_logger.run_id = "run-1"
+        trainer_logger.experiment.get_run.return_value.data.params = params
+        return trainer_logger
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/tests/framework/test_resume_support.py:59]: nope

File: autoware_ml/utils/runtime.py [hunk 2]
```diff
@@ -27,7 +27,8 @@
 import lightning as L
 import torch
 from hydra.core.hydra_config import HydraConfig
-from lightning.pytorch.loggers import Logger
+from lightning.fabric.utilities.logger import _convert_params, _flatten_dict
+from lightning.pytorch.loggers import Logger, MLFlowLogger
 from omegaconf import DictConfig, OmegaConf, open_dict
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/runtime.py:33]: Nope

File: autoware_ml/cli/runtime.py [hunk 3]
```diff
@@ -134,6 +137,45 @@ def temporary_environment(updates: dict[str, str | None]):
...
+            f"No '{RUN_METADATA_FILENAME}' found in the parent directories of "
+            f"'{checkpoint_path}', so the source MLflow run cannot be reused. "
+            "Pass --new-run to resume into a fresh run."
+        )
+    if metadata["config_name"] != config_name:
+        raise ValueError(
+            f"Resume checkpoint belongs to config '{metadata['config_name']}', "
+            f"but '{config_name}' was launched."
+        )
+    run_context = load_run_context(tracking_uri, metadata["run_id"])
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/cli/runtime.py:180]: https://github.com/tier4/autoware-ml/pull/66/commits/7e08e2aa6b3f4d3a0c833e797042a98d444ba2dd

File: autoware_ml/utils/runtime.py [hunk 4]
```diff
@@ -158,6 +159,49 @@ def instantiate_trainer(
...
+    for entry in drifted:
+        logger.warning("Param drift against the run's logged params: %s", entry)
+    if drifted:
+        trainer_logger.experiment.set_tag(
+            trainer_logger.run_id, _PARAM_DRIFT_TAG, "; ".join(drifted)[:_MLFLOW_TAG_VALUE_LIMIT]
+        )
+    return new_params
+
+
 def log_hyperparameters(cfg: DictConfig, trainer_logger: Logger | None, trainer: L.Trainer) -> None:
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/runtime.py]: Part of https://github.com/tier4/autoware-ml/pull/66/commits/aae9e178366c8713a3d0c619599af07ae6bf463a

File: autoware_ml/scripts/train.py [hunk 5]
```diff
@@ -98,6 +99,17 @@ def main(cfg: DictConfig):
...
+        progress = torch.load(resume_checkpoint_path, map_location="cpu", weights_only=False)
+        logger.info(
+            "Resuming from '%s': checkpoint saved at epoch %d (global step %d), "
+            "training continues at epoch %d.",
+            resume_checkpoint_path,
+            progress["epoch"],
+            progress["global_step"],
+            progress["epoch"] + 1,
+        )
+        del progress
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/scripts/train.py:114]: Part of https://github.com/tier4/autoware-ml/pull/66/commits/aae9e178366c8713a3d0c619599af07ae6bf463a

File: autoware_ml/utils/runtime.py [hunk 6]
```diff
@@ -158,6 +159,49 @@ def instantiate_trainer(
...
+    composed configuration's.
+
+    Args:
+        trainer_logger: MLflow logger attached to the current run.
+        params: Sanitized configuration parameters about to be logged.
+
+    Returns:
+        Flattened parameters that are not yet logged on the run.
+    """
+    logged = trainer_logger.experiment.get_run(trainer_logger.run_id).data.params
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/runtime.py:185]: https://github.com/tier4/autoware-ml/pull/66/commits/ac43ac5f1774f79e39e208570258cb1f714dd4be
